### PR TITLE
Tab components

### PR DESCRIPTION
**Pull request by rsek**
relies on #390 

standardizes tab-panel type widgets to a single component. off the top of my head, the vertical tabs from the move creation dialog are exceptions; the new component happily accomodates vertical tabs by using css selectors for `aria-orientation`, but currently only accounts for text labels on the tabs rather than add'l clickables.

i'll think on the best way to handle that - probably adding an additional key to the data object for each tab - and address it in a PR specifically for updating the move creation dialog.

in the mean time, the old tabs work fine, and i'd like to get this merged because some other branches i'm working on depend on the component.

- [ ] Update CHANGELOG.md

### CODE REVIEW

File: src/module/vue/sf-charactersheet.vue [hunk 1]
```diff
@@ -108,18 +98,18 @@
...
-}
-.tablist {
-  border-bottom: 1px solid grey;
-}
-.tab {
-  text-align: center;
-  padding: 5px;
-  &.active {
-    background-color: darkgray;
+<style lang="less">
```
Comment (ben) [inline src/module/vue/sf-charactersheet.vue:101]: Are you not fond of `scoped`? It allows us to avoid having to write CSS classes just to scope styles in a single component.

File: tsconfig.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "checkJs": false,
     "allowSyntheticDefaultImports": true,
     "resolveJsonModule": true,
+    "jsx": "preserve",
```
Comment (ben) [inline tsconfig.json:15]: Huh. Why is this necessary?

**Review by ben — COMMENTED** [review]
This is definitely a good direction. A couple of visual nits.

First, the hover state is really unreadable.

CleanShot 2022-06-25 at 15 53 30

Secondly, there's a 1-pixel border on the top of the move sheet that isn't in the SF character sheet. Is that intentional?

CleanShot 2022-06-25 at 15 54 39

Thirdly, the text style of the tab labels is inconsistent. It's the new underline that breaks the camel's back for me – we don't use that anywhere else, and I don't think it belongs. Could we also make them all-caps, like other headers?

File: src/module/vue/sf-charactersheet.vue (hunk 1 above)
Comment (rsek) [reply src/module/vue/sf-charactersheet.vue:101]: i'm not against `scoped` *per se* so much as `scoped` as a way to avoid CSS classes. here's my rationale on that:

1) i reckon the OuterHTML of *every* component -- plus any major elements within the component -- should have distinct, descriptive CSS classes in the first place to make the code easier to grok. this can be accomplished with comments, too, but adding a descriptive CSS class isn't really any more work.
2) with SFCs, identifying a specific element with a descriptive CSS class makes it easier to tell which styles in the template's `<style>` tag belong to which elements
3) inferring *why* an element is receiving a specific style becomes easier: "this is styled this way because scope 1234 specifies it" is less informative than "this is styled this way because it's a `.foo`", especially when an item is receiving styles from multiple selectors
4) finding where the heck the styling *happens* (especially with commonly-repeated properties) is easier if i can search for a specific class name. admittedly, less of a big deal the component CSS is exported as single files and can be inferred from that, though
5) beyond the semantics, the majority of html elements are just a box you can restyle in any number of ways... and often there's multiple ways of structuring the same element (though some are more semantically logical than others). descriptive css class names makes it easier to rewrite the html while keeping all the styling the same. so if i'm rewriting the `foo.vue`, and it has two things styled as buttons labelled "bar" and "baz", it's easier to rewrite that if i class its various widgets descriptively as `.foo-btn-bar` and `.foo-btn-baz` and target those for styling (and then figure out whether they need to be `<a>`, `<button>`, or something else entirely).

none of this precludes `scoped`, but if the above is happening anyways, it does make scoped less broadly applicable.

File: tsconfig.json (hunk 2 above)
Comment (rsek) [reply tsconfig.json:15]: oh - oops, that's one of the flags used when compiling Vue SFCs from TS. earlier i experimented with doing some incremental TS migration by building the new components with it. after running into trouble with the TSC segfault, i set aside that idea, but evidently i forgot to clean up after.

come to think of it, that might be a good idea to revisit now that ive patched the segfault, but doing that for this + the buttons could be its own PR.

**Comment by rsek** [comment]
> First, the hover state is really unreadable.
> 
> CleanShot 2022-06-25 at 15 53 30

oof, agreed - thought i caught all of those, but i think i got confused about the overlapping interactions with `:hover`, `text`, `block`, and `clickable`. FWIW, there's some other contrast issues on hover effects that might be worth addressing. hovering `.clickable.text` (good example: the "Spirit" roll button next to the spirit meter) falls way short of WCAG contrast recs, for instance. i've been holding off on those in an attempt to be disciplined about PR scope.

long term, it might be worth reorganizing a couple distinct classes (`clickable-text`, `clickable-block`, `clickable-textblock`, for instance, or using some other means of categorizing them entirely), so that the relevant styles are gathered under a single header, rather than trying to tease apart the various interactions there.

**Comment by rsek** [comment]
> CleanShot 2022-06-25 at 15 54 39
> 
> Thirdly, the text style of the tab labels is inconsistent. It's the new underline that breaks the camel's back for me – we don't use that anywhere else, and I don't think it belongs. Could we also make them all-caps, like other headers?

here's the current styling from `main`'s `styles.less`, from which i inferred "underlines are intended (if not currently functional) when a character sheet tab is selected". it just got consolidated to the tabs component, but removing it is no problem.

Screen Shot 2022-06-25 at 5 20 06 PM

**Comment by rsek** [comment]
> Secondly, there's a 1-pixel border on the top of the move sheet that isn't in the SF character sheet. Is that intentional?
> 
> CleanShot 2022-06-25 at 15 54 39

should be fixed now

**Comment by ben** [comment]
✨
